services/storage/src/simcore_service_storage/app_handlers.py

```diff
@@ -0,0 +1,40 @@
+"""
+
+    - Checks connectivity with other services in the backend
+
+"""
+from aiohttp.web import Request, RouteTableDef
+from models_library.app_diagnostics import AppStatusCheck
+
+from .db import get_engine_state
+from .db import is_service_responsive as is_pg_responsive
+from .meta import api_version, api_version_prefix, app_name
+
+# TODO: from .s3 import is_s3_responsive
+
+VX = f"/{api_version_prefix}"
+
+routes = RouteTableDef()
+
+
+@routes.get(VX + "/status")
+async def get_app_status(request: Request):
+    # NOTE: all calls here must NOT raise
+
+    status = AppStatusCheck.parse_obj(
+        {
+            "name": app_name,
+            "version": api_version,
+            "services": {
+                "postgres": {
+                    "healthy": await is_pg_responsive(request.app),
```

Comment:
Dose this have an internal timeout `is_pg_responsive`? Will it return sub 100 ms in all cases? Being a status call I would expect it to be really really fast.

If you have a look at `is_pg_responsive` I would put a timeout on it, and make sure the connection is released properly as well.

Reply:
yes, it uses the ``aiopg.sa.Engine``  client

Reply:
what if the connection to the database is down( this will not timeout very fast). I would also use a timeout on this is_pg_responsive call.

Reply:
OK, I managed to hack ``aiopg.sa`` to get a fast response but while I was manually testing it (I stopped postgres) I noticed that with the database we have still another problem since this entrypoint needs authentication and the latter the database ... Nonetheless, i will solve this separately and created #2260